src/PowerShellEditorServices/Workspace/WorkspaceFileSystemWrapper.cs

```diff
@@ -0,0 +1,317 @@
+        /// </summary>
+        internal IEnumerable<FileSystemInfo> SafeEnumerateFileSystemInfos(DirectoryInfo dirInfo)
+        {
+            IEnumerable<FileSystemInfo> result = null;
+            try
+            {
+                result = dirInfo.EnumerateFileSystemInfos("*", SearchOption.TopDirectoryOnly);
+            }
+            catch (DirectoryNotFoundException e)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to it being an invalid path",
+                    e);
+                yield break;
+            }
+            catch (PathTooLongException e)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to the path being too long",
+                    e);
+                yield break;
+            }
+            catch (Exception e) when (e is SecurityException || e is UnauthorizedAccessException)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to the path not being accessible",
+                    e);
+                yield break;
+            }
+            catch (Exception e)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to an exception",
+                    e);
+                yield break;
+            }
+            if (result == null) { yield break; }
+            foreach (System.IO.FileSystemInfo item in result)
+            {
+                if (_ignoreReparsePoints && (item.Attributes & FileAttributes.ReparsePoint) != 0) { continue; }
+                // If it's a directory then automatically return it
+                if ((item.Attributes & FileAttributes.Directory) != 0) { yield return item; continue; }
+                if (_allowedExtensions == null || _allowedExtensions.Length == 0) { yield return item; continue; }
+                foreach (string extension in _allowedExtensions)
+                {
+                    if (item.Extension == extension) { yield return item; break; }
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Wraps an instance of <see cref="System.IO.DirectoryInfo" /> and provides implementation of
+    /// <see cref="Microsoft.Extensions.FileSystemGlobbing.Abstractions.DirectoryInfoBase" />.
+    /// Based on https://github.com/aspnet/Extensions/blob/c087cadf1dfdbd2b8785ef764e5ef58a1a7e5ed0/src/FileSystemGlobbing/src/Abstractions/DirectoryInfoWrapper.cs
+    /// </summary>
+    public class WorkspaceFileSystemDirectoryWrapper : DirectoryInfoBase
+    {
+        private readonly DirectoryInfo _concreteDirectoryInfo;
+        private readonly bool _isParentPath;
+        private readonly WorkspaceFileSystemWrapperFactory _factory;
```

Comment:
```suggestion
        private readonly WorkspaceFileSystemWrapperFactory _fsWrapperFactory;
```

Since this is a field, a descriptive name will go a long way

Reply:
Yeah I suppose.  Because it's a private method, I saw it's scope was small and didn't need a full name.  Will change.

Reply:
Done